manuscripts2/report.py

```diff
@@ -81,69 +90,185 @@ def __init__(self, es_url=None, start=None, end=None, data_dir=None, filters=Non
-                                new_config[section][metric_section] is None):
-                                new_config[section][metric_section] = ds_config[section][metric_section]
-                            else:
-                                new_config[section][metric_section] += ds_config[section][metric_section]
-
-            activity_metrics = ds_config['project_activity']['metrics']
-            new_config['project_activity']['ds' + str(index + 1) + "_metrics"] = activity_metrics
-
-        # Fields that are not linked to a data source
-        new_config['overview']['activity_file_csv'] = "data_source_evolution.csv"
-        new_config['overview']['efficiency_file_csv'] = "efficiency.csv"
-        new_config['project_process']['time_to_close_title'] = "Days to close (median and average)"
-        new_config['project_process']['time_to_close_review_title'] = "Days to close review (median and average)"
-
-        return new_config
-
-    def get_activity_metrics(self):
-
-        metrics = self.config['overview']['activity_metrics']
-        file_name = self.config['overview']['activity_file_csv']
-        data_path = os.path.join(self.data_dir, "data")
+        data_path = os.path.join(self.data_dir, "overview")
         if not os.path.exists(data_path):
             os.makedirs(data_path)
-        file_name = os.path.join(data_path, file_name)
 
-        logger.debug("CSV file %s generation in progress", file_name)
+        overview_config = {
+            "activity_metrics": [],
+            "author_metrics": [],
+            "bmi_metrics": [],
+            "time_to_close_metrics": [],
+            "projects_metrics": []
+        }
+
+        for ds in self.data_sources:
+            metric_file = self.ds2class[ds]
+            metric_index = self.get_metric_index(ds)
+            overview = metric_file.overview(metric_index, self.start_date, self.end_date)
+            for section in overview_config:
+                overview_config[section] += overview[section]
+
+        overview_config['activity_file_csv'] = "data_source_evolution.csv"
+        overview_config['efficiency_file_csv'] = "efficiency.csv"
+
+        # ACTIVITY METRICS
+        metrics = overview_config['activity_metrics']
+        file_name = overview_config['activity_file_csv']
+        file_name = os.path.join(data_path, file_name)
 
         csv = "metricsnames, netvalues, relativevalues, datasource\n"
 
         for metric in metrics:
-            (last, percentage) = get_trend(metric.get_timeseries())
-            csv += "{}, {}, {}, {}\n".format(metric.index.index_name, last,
-                                             percentage, metric.index.index_name)
+            (last, percentage) = get_trend(metric.timeseries())
+            csv += "{}, {}, {}, {}\n".format(metric.name, last,
+                                             percentage, metric.id)
+        create_csv(file_name, csv)
 
+        # AUTHOR METRICS
```

Comment:
I think we should have some common code for doing this kind of stuff. Apparently, this is just creating tables, figs, from the metrics, and apparently (given the proper label, filename, etc as arguments) this could be implemented with a function, that could be called for whatever the metrics that has this structure. Am I right?

Reply:
I haven't gotten into the other sections of the report. But yes, I think we can create a function out of this.

Reply:
Please, do.

Reply:
I want to wait until I got to the other sections of the report so that the function can be generalised better. Is that okay with you?

Reply:
OK, but please, remember this... Now, the code is becoming too spaghetti...